vmdb/app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -0,0 +1,140 @@
+module EmsRefresh::Parsers
+  class Kubernetes
+    def self.ems_inv_to_hashes(inventory)
+      new.ems_inv_to_hashes(inventory)
+    end
+
+    def initialize
+      @data = {}
+    end
+
+    def ems_inv_to_hashes(inventory)
+      get_nodes(inventory)
+      get_services(inventory)
+      get_pods(inventory)
+      @data
+    end
+
+    def get_nodes(inventory)
+      process_collection(inventory["node"], :container_nodes) { |n| parse_node(n) }
+    end
+
+    def get_services(inventory)
+      process_collection(inventory["service"], :container_services) { |s| parse_service(s) }
+    end
+
+    def get_pods(inventory)
+      process_collection(inventory["pod"], :container_groups) { |n| parse_pod(n) }
+    end
+
+    def process_collection(collection, key, &block)
+      @data[key] ||= []
+      collection.each { |item| process_collection_item(item, key, &block) }
+    end
+
+    def process_collection_item(item, key)
+      @data[key] ||= []
+
+      new_result = yield(item)
+
+      @data[key] << new_result
+      new_result
+    end
+
+    def parse_node(node)
+      new_result = parse_base_item(node)
+      # TODO, add more properties such as resources
+      new_result
+    end
+
+    def parse_service(service)
+      new_result = parse_base_item(service)
+      new_result[:port] = service.spec.port
+      new_result[:protocol] = service.spec.protocol
+      new_result[:portal_ip] = service.spec.portalIP
+      new_result[:container_port] = service.spec.containerPort
+      new_result[:namespace] = service.metadata.instance_values["table"][:namespace]
+      new_result[:session_affinity] = service.spec.sessionAffinity
+      new_result
```

Comment:
This can be cleaned up a bit by use .merge:

```ruby
new_result = parse_base_item(service)
new_result.merge(
  :port             => service.spec.port
  :protocol         => service.spec.protocol
  :portal_ip        => service.spec.portalIP
  :container_port   => service.spec.containerPort
  :namespace        => service.metadata.instance_values["table"][:namespace]
  :session_affinity => service.spec.sessionAffinity
)
```